packages/genesys-spark-components/src/components/stable/gux-listbox/option-group/gux-option-group.tsx

```diff
@@ -9,17 +12,103 @@ import { Component, h, Host, JSX, Prop } from '@stencil/core';
   shadow: true
 })
 export class GuxOptionGroup {
+  private rootParent: HTMLElement;
+  private parentObserver: MutationObserver;
+
+  @Element() root: HTMLGuxOptionGroupBetaElement;
+
   @Prop()
   label!: string;
 
   // @Prop()
   // disabled: boolean;
   // TOOD: Add disabled state once design work is complete
 
+  @State()
+  private parentFilterValue: string = '';
+
+  @State()
+  filtered: boolean = false;
+
+  @State()
+  hideDivider: boolean = false;
+
+  componentDidLoad(): void {
+    this.rootParent = getClosestElement(
+      'gux-dropdown',
+      this.root
+    ) as HTMLElement;
+
+    this.parentObserver = onMutation(this.rootParent, () => {
+      this.hideDivider =
+        this.parentFilterValue.length > 0 &&
+        !this.hasVisibleGroupAfter(this.root);
+    });
+
+    if (this.rootParent) {
+      this.rootParent.addEventListener(
+        'guxfilter',
+        this.handleFilterChange.bind(this) as EventListenerOrEventListenerObject
+      );
+    }
+  }
+
+  disconnectedCallback(): void {
+    if (this.rootParent) {
+      this.rootParent.removeEventListener(
+        'guxfilter',
+        this.handleFilterChange.bind(this) as EventListenerOrEventListenerObject
+      );
+      this.parentObserver.disconnect();
+    }
+  }
+
+  private isOption(item: Element): boolean {
+    const optionTypes = ['GUX-OPTION', 'GUX-OPTION-ICON'];
+    return optionTypes.includes(item.tagName);
+  }
+
+  handleFilterChange(event: CustomEvent<string>) {
+    this.parentFilterValue = event.detail;
+
+    const options = Array.from(this.root?.children).filter(child =>
+      this.isOption(child)
+    ) as ListboxOptionElement[];
+    const allOptionsFiltered = !options.some(option =>
+      this.matchOption(option, this.parentFilterValue)
+    );
+
+    this.filtered = this.parentFilterValue.length > 0 && allOptionsFiltered;
```

Comment:
I don't think this is the right way to do this. The application is responsible for what is filtered. You have to look at the options `filtered` property not do the filtering here.

Reply:
Had a look at that again, have done it that way now. Thanks